### PR TITLE
[URL] Add tests for the distinct percent-encode sets

### PR DESCRIPTION
**Pull request by alwinb**
This adds a number of test cases to test the distinct percent-encode sets as specified in the [Percent-encoded bytes][1] section of the WHATWG URL Standard. 

[1]:https://url.spec.whatwg.org/#percent-encoded-bytes

### CODE REVIEW

File: url/resources/urltestdata.json [hunk 1]
```diff
@@ -7429,5 +7429,185 @@
...
+    "username": "%20!%22$%&'()*+,-.%3B%3C%3D%3E%40%5B%5C%5D%5E_%60%7B%7C%7D~"
+  },
+  {
+    "input": "wss:// !\"$%&'()*+,-.;<=>@[]^_`{|}~@host/",
+    "base": "about:blank",
+    "hash": "",
+    "host": "host",
+    "hostname": "host",
+    "href": "wss://%20!%22$%&'()*+,-.%3B%3C%3D%3E%40%5B%5D%5E_%60%7B%7C%7D~@host/",
+    "origin": "null",
```
Comment (domenic) [inline url/resources/urltestdata.json:7456]: jsdom/whatwg-url says this should be `wss://host`

File: url/resources/urltestdata.json [hunk 2]
```diff
@@ -7429,5 +7429,185 @@
...
+    "username": "joe"
+  },
+  {
+    "input": "wss://joe: !\"$%&'()*+,-.:;<=>@[]^_`{|}~@host/",
+    "base": "about:blank",
+    "hash": "",
+    "host": "host",
+    "hostname": "host",
+    "href": "wss://joe:%20!%22$%&'()*+,-.%3A%3B%3C%3D%3E%40%5B%5D%5E_%60%7B%7C%7D~@host/",
+    "origin": "null",
```
Comment (domenic) [inline url/resources/urltestdata.json:7486]: jsdom/whatwg-url says this should be `wss://host`

**Review by domenic — COMMENTED** [review]
Found a couple mismatches between this and jsdom/whatwg-url. Whether it's the tests that are wrong, or jsdom/whatwg-url, is unclear... let me know what you think.

**Comment by annevk** [comment]
They have an origin per https://url.spec.whatwg.org/#origin. Thanks for working on this @alwinb! Are you also planning to file browser bugs? I suspect some of these might be tracked already, but it might help to have a fresh one with all the relevant data in it.

**Comment by alwinb** [comment]
No you're right, the origin should be `wss://host` in those cases. I checked locally in the browser (via wpt serve) and did not spot this. This is why extra eyes are a good thing!

**Review by domenic — APPROVED** [review]
LGTM, all tests pass with jsdom/whatwg-url.

**Comment by alwinb** [comment]
@annevk I am a bit uneasy about filing browser bugs. I wonder if more will come up while I study this, and there are some issues that look like they will trigger additional small changes to the encode sets. I kind of prefer to send a single bug report to them rather than several with changes in a row. Maybe that's being too careful, but still...

**Comment by annevk** [comment]
Thanks for the update, I think that's fine in this case.

**Pull request merged**